src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonPropertyInfo.cs

```diff
@@ -448,7 +448,7 @@ internal void Configure()
 
             if (IsRequired)
             {
-                if (!CanDeserialize)
+                if (!CanDeserialize && AssociatedParameter?.IsRequiredParameter != true)
```

Comment:
should you also be checking for HasDefaultValue here?

Reply:
The `IsRequiredParameter` property that takes that into account (as well as the global `options` setting)

Reply:
I'd appreciate tests around this being properly/incorrectly configured (or did I miss it somehow?)

Reply:
Even though the `JsonParameterInfo` has been made public for use by JSON schema, the public APIs are read-only. From a contract customization perspective the mutable API surface remains the same. In https://github.com/dotnet/runtime/pull/102499 and https://github.com/dotnet/runtime/pull/102902 I added a few tests checking that this API surface is properly invalidated (e.g. in cases where properties are removed or the constructor delegate is replaced).

Reply:
Unless you mean somebody setting incorrect values in the [parameter models used by the source generator](https://learn.microsoft.com/en-us/dotnet/api/system.text.json.serialization.metadata.jsonparameterinfovalues?view=net-8.0)? We've never validated those, which is probably ok since they're marked `EditorBrowsable.Never`.